http/transport-component.md

```diff
@@ -0,0 +1,121 @@
+# HTTP Transport Component <!-- omit in toc -->
+
+| Lifecycle Stage | Maturity      | Status | Latest Revision |
+| --------------- | ------------- | ------ | --------------- |
+| 1A              | Working Draft | Active | r0, 2023-05-31  |
+
+Authors: [@marcopolo]
+
+Interest Group: [@marcopolo], [@mxinden], [@marten-seemann]
+
+[@marcopolo]: https://github.com/marcopolo
+[@mxinden]: https://github.com/mxinden
+[@marten-seemann]: https://github.com/marten-seemann
+
+## Table of Contents <!-- omit in toc -->
+- [Context](#context)
+- [What is an HTTP transport](#what-is-an-http-transport)
+- [Multiaddr representation](#multiaddr-representation)
+- [HTTP Paths](#http-paths)
+
+
+## Context
+
+This document is only about advertising support for an HTTP transport. It
+doesn't make any assertions about how libp2p should interact with that
+transport. That will be defined in a future document.
+
+This exists to clarify the role of the `/http` component in Multiaddrs early to
+avoid confusion and conflicting interpretations.
+
+## What is an HTTP transport
+
+An HTTP transport is simply a node that can speak some standardized version of
+HTTP that is at least HTTP/1.1. Currently that means HTTP/1.1, HTTP/2, and
```

Comment:
Is it necessary to specify this? HTTP/1.0 is not really in use any more, but it would be nice if we didn't have to say anything here.

Reply:
Yeah, I can remove the list. We should still specify at least 1.1.

Reply:
Why?

Reply:
Because HTTP/1.0 is still in use (It's the default for nginx: https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_http_version). We should be explicit that we only support at least 1.1

Reply:
That seems like an argument for not specifying it. Let the HTTP layer do the HTTP things.

Reply:
Do you want to support HTTP/1.0? I don't. If we don't set a minimum requirement we are effectively saying we support HTTP/1.0

Reply:
We’re not. We’re just saying we don’t get involved in HTTP internals. How you configure your HTTP client and server is on you.

Reply:
* We're defining a standard here that folks will reference when they want their nodes to interoperate with the network.
* We don't want them to use HTTP/1.0.

What parts do you disagree with?

Reply:
We’re also not telling people which cipher suites they need to support when using TLS / QUIC. Somebody could configure some obscure cipher suites, which would make cipher suite negotiation fail during the handshake.

Reply:
I don't feel strongly on this. I will remove the HTTP/1.1 requirement.